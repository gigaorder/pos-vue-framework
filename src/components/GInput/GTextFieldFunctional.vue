<template>
  <div class="tf-wrapper" :class="tfClasses" @click="onClick" @mouseup="onMouseUp" @mousedown="onMouseDown">
    <div class="tf-prepend__outer"></div>
    <fieldset>
      <div class="tf">
        <div class="tf-prepend__inner">
          <slot></slot>
        </div>
        <span ref="prefixRef" class="tf-affix">{{prefix}}</span>
        <input id="input" type="text"
               class="tf-input"
               :type="type"
               :label="label"
               v-model="internalValue"
               :placeholder="placeholder"
               ref="input"
               @change="onChange"
               @focus="onFocus"
               @blur="onBlur"
               @keydown="onKeyDown">
        <label for="input" class="tf-label" :class="labelClasses" :style="labelStyles">
          <slot name="label">{{label}}</slot>
        </label>
        <span class="tf-affix">{{suffix}}</span>
        <div class="tf-append__inner">
          <slot></slot>
        </div>
      </div>
    </fieldset>
    <div class="tf-append__outer" @click="onClearIconClick">
      <slot></slot>
      <img v-if="isDirty && clearable" src="../../assets/delivery/cancel.svg">
    </div>
    <div class="tf-error" v-if="!isValidInput">{{errorMessage}}</div>
    <div class="tf-hint" v-else-if="!solo" :class="hintClasses">{{hint}}</div>
    <div v-show="counter" class="tf-counter">{{counterValue}} / {{maxlength}}</div>
  </div>
</template>

<script>
  import {computed, ref, onMounted} from '@vue/composition-api';
  import getGInput from "./GInput";
  import getGInputField from "./GInputField";
  import {keyCodes} from '../../utils/helpers';

  export default {
    name: "GTextFieldFunctional",
    props: {
      ...{
        label: String,
        disabled: Boolean,
        readOnly: Boolean,
        appendIcon: String,
        prependIcon: String,
        prefix: String,
        suffix: String,
      },
      required: Boolean,
      placeholder: String,
      clearable: Boolean,
      rules: Function,
      hint: String,
      persistent: Boolean,
      errorMessage: String,
      counter: Boolean,
      maxlength: String,

      // display
      large: Boolean,
      textColor: {
        type: String,
        default: '#00acc1'
      },
      bordered: Boolean,
      centered: Boolean,
      active: Boolean,

      //styles
      filled: Boolean,
      outlined: Boolean,
      solo: Boolean,
      shaped: Boolean,
      rounded: Boolean,
      flat: Boolean,

      // basic props
      value: [String, Number],
      type: {
        type: String,
        default: 'text',
      },

    },
    setup(props, context) {
      const {listeners} = getGInput(props, context)
      const {
        counterValue,
        isDirty,
        isLabelActive,
        isFocused,
        hasIcon,
        onClick,
        onFocus,
        onBlur,
        internalValue,
        isValidInput,
        onClearIconClick,
        prependClasses,
        tfClasses,
        hintClasses,
        labelStyles,
        labelClasses,
        prefixRef,
      } = getGInputField(props, context)

      // template refs
      const input = ref(null)


      // event listeners
      const {onChange} = listeners

      function onKeyDown(event) {
        if (event.keyCode === keyCodes.enter && props.isDirty()) {
          context.emit('change', internalValue.value);
        }
        context.emit('keydown', event)
      }

      function onMouseUp(event) {
        listeners.onMouseUp(event);
        onFocus();
      }

      function onMouseDown(event) {
        listeners.onMouseDown(event);
        if (event.target !== context.refs.input) {
          event.preventDefault();
          event.stopPropagation();
        }
      }

      return {
        prefixRef,
        labelClasses,
        labelStyles,
        prependClasses,
        tfClasses,
        hintClasses,
        internalValue,
        counterValue,
        isLabelActive,
        onClick,
        onChange,
        onFocus,
        onBlur,
        isFocused,
        onKeyDown,
        onMouseUp,
        onMouseDown,
        onClearIconClick,
        isDirty,
        isValidInput,
        hasIcon,
      }
    }
  }
</script>

<style scoped>
  @import "_GInputField.scss";
</style>
