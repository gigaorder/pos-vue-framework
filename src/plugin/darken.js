import {getHslColor} from "../utils/colors";

;(function (window) {
  'use strict';

  const Darken = window.Darken || {};

  const Effect = {
    //click
    elements: [],
    // Effect delay
    duration: 1500,

    show: function (e, element) {

      // Disable right click
      if (e.button === 2) {
        return false;
      }

      const el = element || this;
      Effect.elements.push(el)

      const background = el.style.backgroundColor,
          border = el.style.borderColor,
          shadow = el.style.boxShadow

      let {h, s, l, a} = getHslColor(background)
      if (h === 0 && s === 0) {
        let hsla
        if (border)
          hsla = getHslColor(border)
        else
          hsla = {h: 0, s: 0, l: 0, a: 0}
        h = hsla.h
        s = hsla.s
        l = hsla.l
        a = hsla.a
      }
      el.style.background = `hsla(${h}, ${s - 5}%, ${l - 15}%, ${a})`
      el.style.borderColor = `hsla(${h + 2}, ${s - 10}%, ${l - 15}%, ${a})`
      el.style.boxShadow = `1px 0px 5px hsla(${h + 2}, ${s - 10}%, ${l - 15}%, ${a})`;

      setTimeout(() => {
        el.style.backgroundColor = background
        el.style.borderColor = border
        el.style.boxShadow = shadow
        const index = Effect.elements.findIndex(element => element === el)
        Effect.elements.splice(index, 1)
      }, Effect.duration)
    },

  };

  const TouchHandler = {
    touches: 0,
    allowEvent: function (e) {
      let allow = true;

      if (e.type === 'touchstart') {
        TouchHandler.touches += 1;
      } else if (e.type === 'touchend' || e.type === 'touchcancel') {
        setTimeout(function () {
          if (TouchHandler.touches > 0) {
            TouchHandler.touches -= 1;
          }
        }, Effect.duration);
      } else if (e.type === 'mousedown' && TouchHandler.touches > 0) {
        allow = false;
      }

      return allow;
    },
    touchup: function (e) {
      TouchHandler.allowEvent(e);
    }
  };

  function getWavesEffectElement(e) {
    if (TouchHandler.allowEvent(e) === false) {
      return null;
    }

    let element = null;
    let target = e.target || e.srcElement;

    while (target.parentNode !== null) {
      if (!(target instanceof SVGElement) && target.className.indexOf('darken-effect') !== -1) {
        element = target;
        break;
      }
      target = target.parentNode;
    }
    return element;
  }

  function showEffect(e) {
    const element = getWavesEffectElement(e);

    if (element !== null && Effect.elements.findIndex(el => el === element) === -1) {
      Effect.show(e, element);
    }
  }

  Darken.displayEffect = function (options) {
    options = options || {};

    if ('duration' in options) {
      Effect.duration = options.duration;
    }


    if ('ontouchstart' in window) {
      document.body.addEventListener('touchstart', showEffect, false);
    }

    document.body.addEventListener('mousedown', showEffect, false);
  };

  Darken.attach = function (element) {
    if ('ontouchstart' in window) {
      element.addEventListener('touchstart', showEffect, false);
    }

    element.addEventListener('mousedown', showEffect, false);
  };

  window.Darken = Darken;

  document.addEventListener('DOMContentLoaded', function () {
    Darken.displayEffect();
  }, false);

})(window);
