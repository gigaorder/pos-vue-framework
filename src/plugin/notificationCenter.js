import Vue from 'vue';
import GNotification from '../components/GNotification/GNotification';

export default {
  install(Vue, options) {
    const notificationCenter = new Vue({
      ...GNotification,
      propsData: options ? parseOptions(options) : {}
    })
    notificationCenter.$mount(document.body.appendChild(document.createElement('div')));

    Vue.prototype.$notificationCenter = notificationCenter;
  }
}

function parseOptions(options) {
  const propsData = {}

  if(options.position) {
    const position = options.position.split('-');
    for (let i = 0; i < position.length; i++) {
      propsData[position[i]] = true
    }
  }

  if (options.width) propsData['width'] = options.width

  return propsData;
}