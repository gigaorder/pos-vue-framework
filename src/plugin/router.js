import Vue from 'vue';
import VueRouter from 'vue-router';
import KeyboardDemo from '@/view/KeyboardDemo';
import Groupable from '@/view/Groupable';
import MenuDemo from '@/view/MenuDemo';
import SliderDemo from '../view/SliderDemo';
import OverlayDemo from "../view/OverlayDemo";
import SelectIonControl from '../view/SelectIonControl';
import DialogDemo from '../view/DialogDemo';
import PulseDemo from '../view/PulseDemo';
import ToolTipDemo from '../view/ToolTipDemo';
import InputDemo from "../view/InputDemo";
import TransitionDemo from '../view/TransitionDemo';
import ListDemo from '../view/ListDemo';
import List2Demo from "../view/List2Demo";
import DndDialogDemo from '../view/DndDialogDemo';
import GChipGroupDemo from '../view/GChipGroupDemo';
import DatePicker from '../view/DatePickerDemo';
import ExpansionPanelDemo from '../view/ExpansionPanelDemo';
import Textarea from "../view/TextareaDemo";
import LoginView from '../view/LoginView';

const routes = [
  {path: '/login', name: 'Login', component: LoginView},
];

const router = new VueRouter({
  mode: 'history',
  base: '/',
  routes
});

export default router
