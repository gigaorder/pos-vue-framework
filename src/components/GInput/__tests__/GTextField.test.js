import { createElement, setInputValue } from '../../../tests/setup'

const Vue = require('vue/dist/vue.common.js')
import plugin from '@vue/composition-api'

Vue.use(plugin)

// Import your components after all of above import
import GTextFieldFunctional from '../GTextFieldFunctional';


describe('GTextField', function () {
  afterEach(() => {
    // clear document body after each test to prevent content slots from
    // mounting on the first div with [data-app] attr
    document.body.innerHTML = ''
  })

  it('validate test', async function (done) {
    let vm = new Vue({
      template: `<g-text-field-functional label="Has prefix"
                               filled
                               clearable
                               counter="25"
                               prefix="prefix"
                               persistent
                               v-model="testValue"
                               hint="This is persistent hint"
                               :rules="[rules.required, rules.counter]"
                              
      >
        <template v-slot:prepend-inner >
          <img src="../../../assets/delivery/avatar.svg" alt="avatar">
        </template>
        <template v-slot:prepend-outer >
          <img src="../../../assets/delivery/avatar.svg" alt="avatar">
        </template>
        <template v-slot:append-inner >
          <img src="../../../assets/delivery/avatar.svg" alt="avatar">
        </template>
        <template v-slot:append-outer >
          <img src="../../../assets/delivery/avatar.svg" alt="avatar">
        </template>
      </g-text-field-functional>`,
      components: { GTextFieldFunctional },
      data: () => {
        return {
          testValue: null,
          rules: {
            required: value => !!value || 'Required.',
            counter: value => value.length > 4 || 'Min 5 characters',
            email: value => {
              const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
              return pattern.test(value) || 'Invalid e-mail.'
            }
          }
        }
      }
    }).$mount();

    //vm.testValue = 'test';

    setInputValue(vm.$el.querySelector('input'), 'test');
    await vm.$nextTick() && await vm.$nextTick();

    expect(vm.$el.outerHTML).toMatchSnapshot();
    done();
  })
  it('should render wrapper', function () {
    let vm = new Vue({
      template:`<g-text-field-functional filled></g-text-field-functional>`,
      components:{ GTextFieldFunctional}
    }).$mount() ;
    expect(vm.$el.getElementsByClassName('tf__filled')).toBeTruthy()
  })
  it('should render hint, activate, deactivate hint', function () {
    let vm = new Vue({
      template:`<g-text-field-functional hint="Test hint"></g-text-field-functional>`,
      components:{ GTextFieldFunctional}
    }).$mount() ;
    const input = vm.$el.querySelector('input');
    expect(vm.$el.querySelector('.tf-hint')).toBeTruthy()
    input.click();
    vm.$nextTick(() => {
      expect(vm.$el.querySelector('.tf-hint__active')).toBeTruthy();
      input.blur();
      vm.$nextTick(() => {
        expect(vm.$el.querySelector('.tf-hint__active')).toBeFalsy()
      })
    });



  });
  ;
  it('should render label, transform label when input focused', function () {
    let vm = new Vue({
      template:`<g-text-field-functional label="Test label"></g-text-field-functional>`,
      components:{ GTextFieldFunctional}
    }).$mount() ;
    const input = vm.$el.querySelector('input');
    expect(input.getAttribute('label')).toBe('Test label');
    const label = vm.$el.querySelector('.tf-label');
    expect(label).toBeTruthy();
    input.click();
    vm.$nextTick(() => {
      expect(label.style).toContain('transform');
      input.blur();
      vm.$nextTick(() =>{
        expect(label.style).not.toContain('transform');
      })
    })
  });
  it('should render prefix, transform label align to prefix', function () {
    let vm = new Vue({
      template:`<g-text-field-functional label="Test label" prefix="Prefix"></g-text-field-functional>`,
      components:{ GTextFieldFunctional}
    }).$mount();
    const prefix = vm.$el.querySelector('.tf-affix');
    expect(prefix).toBeTruthy();
    expect(prefix.innerHTML).toBe('Prefix');
    expect(vm.$el.outerHTML).toMatchSnapshot();
    const label = vm.$el.querySelector('label');
    label.click();
    vm.$nextTick(() =>{
      expect(vm.$el.querySelector('.tf-hint__active')).toBeTruthy()
    })

  });
  ;
});
