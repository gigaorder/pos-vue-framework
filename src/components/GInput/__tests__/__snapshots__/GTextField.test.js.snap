// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTextField should render test 1`] = `
<div class="tf-wrapper tf__filled tf-wrapper__error">
  <div class="tf-prepend__outer">
    <img src="../../../assets/delivery/avatar.svg"
         alt="avatar"
    >
  </div>
  <fieldset>
    <legend>
    </legend>
    <div class="tf tf__error">
      <div class="tf-prepend__inner">
        <img src="../../../assets/delivery/avatar.svg"
             alt="avatar"
        >
      </div>
      <div class="tf-affix">
        prefix
      </div>
      <div class="inputGroup">
        <input id="input"
               type="text"
               label="Has prefix"
               class="tf-input"
               style="color: red;"
        >
        <label for="input"
               class="tf-label"
               style="transform: translateY(-18px) translateX(0px)  scale(0.75); color: red;"
        >
          Has prefix
        </label>
      </div>
      <div class="tf-affix">
      </div>
      <div class="tf-append__inner">
        <img src="../../../assets/delivery/avatar.svg"
             alt="avatar"
        >
      </div>
    </div>
  </fieldset>
  <div class="tf-append__outer">
    <img src="../../assets/delivery/cancel.svg">
    <img src="../../../assets/delivery/avatar.svg"
         alt="avatar"
    >
  </div>
  <div class="tf-error">
    Min 5 characters.
  </div>
  <div class="tf-counter">
    4 / 25
  </div>
</div>
`;
